### DISCUSSION TITLE
Developer's Mode

### DISCUSSION THREAD

**Discussion by eric-horodyski** [ionic-team/capacitor#4630] [Ideas]
While working with a member of the community, an idea was brought to my attention to provide a "Developer's Mode" that would provide development information, such as network requests and Capacitor logs, on-device as the application is running. 

This mode could be triggered by first setting a Capacitor configuration variable to first enable the ability to access "Developer's Mode" (for non-production environments), then performing a gesture (such as three finger swipe).

**Comment by jcesarmobile** [comment]
Capacitor had a dev mode, I think it was removed on Capacitor 2 because caused more confusion than help. It was simpler than what you propose, I think it only showed JavaScript errors.
